assets/src/edit-story/components/library/libraryProvider.js

```diff
@@ -19,31 +19,39 @@
  */
 import PropTypes from 'prop-types';
 import { useMemo, useState } from 'react';
+import { useFeatures } from 'flagged';
 
 /**
  * Internal dependencies
  */
 import { useInsertElement } from '../canvas';
 import Context from './context';
 
-const MEDIA = 'media';
-const TEXT = 'text';
-const SHAPES = 'shapes';
-const ELEMENTS = 'elements';
-const ANIMATION = 'animation';
-
-const TABS = {
-  MEDIA,
-  TEXT,
-  SHAPES,
-  ELEMENTS,
-  ANIMATION,
+export const TAB_IDS = {
+  MEDIA: 'media',
+  MEDIA3P: 'media3p',
+  TEXT: 'text',
+  SHAPES: 'shapes',
+  ELEMENTS: 'elements',
+  ANIMATION: 'animation',
 };
 
 function LibraryProvider({ children }) {
-  const [tab, setTab] = useState(MEDIA);
+  const [tab, setTab] = useState(TAB_IDS.MEDIA);
   const insertElement = useInsertElement();
 
+  const { showAnimationTab, media3pTab } = useFeatures();
+
+  // Order here is important, as it denotes the actual visual order of elements.
+  const tabs = [
```

Comment:
Add a useMemo to avoid re-renders on components that depend on 'tabs':

const tabs = useMemo(() => [
...
].filter(Boolean), media3pTab, showAnimationTab);

Reply:
Done.